source/codegen/templates/service.cpp.mako

```diff
@@ -62,6 +64,17 @@ namespace ${config["namespace_component"]}_grpc {
       output->AddAllocated(message);
     }
   }
+% endif
+% if custom_type["name"] in input_custom_types:
+   ${custom_type["name"]} ${service_class_prefix}Service::GetStructFromGrpcType(const ${namespace_prefix}${custom_type["grpc_name"]}& input) 
+  {
+    ${custom_type["name"]}* output = new ${custom_type["name"]}();  
+%     for field in custom_type["fields"]: 
+    output->${common_helpers.pascal_to_camel(common_helpers.snake_to_pascal(field["grpc_name"]))} = input.${common_helpers.camel_to_snake(field["name"])}();
+%     endfor
+    return *output;
+  }
```

Comment:
Add a blank line after this and remove the one that is before `endfor` on line 78 below.

Reply:
Done